cannon/mipsevm/versions/detect_test.go

```diff
@@ -47,28 +46,38 @@ func TestDetectVersion(t *testing.T) {
 			})
 		}
 	}
+}
 
-	// Additionally, check that the latest supported versions write new states in a way that is detected correctly
-	t.Run("SingleThreadedBinary", func(t *testing.T) {
-		state, err := NewFromState(singlethreaded.CreateEmptyState())
-		require.NoError(t, err)
-		path := writeToFile(t, "state.bin.gz", state)
-		version, err := DetectVersion(path)
-		require.NoError(t, err)
-		require.Equal(t, VersionSingleThreaded2, version)
-	})
+// Check that the latest supported versions write new states in a way that is detected correctly
+func TestDetectVersion_singleThreadedBinary(t *testing.T) {
+	targetVersion := VersionSingleThreaded2
+	if !arch.IsMips32 {
+		t.Skip("Single-threaded states are not supported for 64-bit VMs")
+	}
 
-	t.Run("MultiThreadedBinary", func(t *testing.T) {
-		state, err := NewFromState(multithreaded.CreateEmptyState())
-		require.NoError(t, err)
-		path := writeToFile(t, "state.bin.gz", state)
-		version, err := DetectVersion(path)
-		require.NoError(t, err)
-		require.Equal(t, VersionMultiThreaded, version)
-	})
+	state, err := NewFromState(singlethreaded.CreateEmptyState())
+	require.NoError(t, err)
+	path := writeToFile(t, "state.bin.gz", state)
+	version, err := DetectVersion(path)
+	require.NoError(t, err)
+	require.Equal(t, targetVersion, version)
+}
+
+func TestDetectVersion_multiThreadedBinary(t *testing.T) {
+	targetVersion := VersionMultiThreaded
+	if !arch.IsMips32 {
+		targetVersion = VersionMultiThreaded64
```

Comment:
`DetectVersion` is used by `multicannon`, which does not contain the `cannon64` build tag.
As such, the function must reliably detect all state versions when `arch.IsMips32 == true`. This test needs to run regardless of `cannon32` such that, `multicannon` can detect multi-threaded 64-bit state versions.

Reply:
We have `TestDetectVersion_fromFile` that runs across all types / architectures.  But for these tests, it creates a new empty state in memory:
```
state, err := NewFromState(multithreaded.CreateEmptyState())
```
So, the type of that state depends on the current `arch` setting.  

Not sure what else to do here.  We could try generating the state by creating a CLI command that generates an empty state and then run the command externally?

Reply:
Ignore my comment; I missed that this is creating a state from memory. This functionality is only used by the cannon-impl programs. The multicannon usecase only detects versions from a state file rather than from memory. So `TestDetectVersion_fromFile` already covers all multicannon cases.